### PR TITLE
Slider hook part 1.

### PR DESCRIPTION
**Pull request by joschect**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ yarn change`

#### Description of changes

Part1 of moving state out of slider into a useSlider hook. I'm intentionally leaving the slide, thumb, slide value, for the next iteration to help reduce the amount that is happening in this change.
#ElectricSlide

### CODE REVIEW

File: packages/react-next/src/components/Slider/Slider.base.tsx [hunk 1]
```diff
@@ -28,274 +23,82 @@ const useComponentRef = (props: ISliderProps, thumb: React.RefObject<HTMLSpanEle
     [value],
   );
 };
+const SliderLabel = (props: {
+  className?: string;
+  label?: string;
+  ariaLabel?: string;
```
Comment (jurokapsiar) [inline packages/react-next/src/components/Slider/Slider.base.tsx:29]: as discussed in https://hackmd.io/k2eJKI_JRs2Gew9dbrz00A?both, can you pls use dashed aria props? See https://reactjs.org/docs/dom-elements.html. camelCased can cause inconsistencies when spreading unhandled props to the root element.

File: packages/react-next/src/components/Slider/useSlider.ts [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+    min = 0,
+    showValue = true,
+    buttonProps = {},
+    vertical = false,
+    valueFormat,
+    styles,
+    theme,
+  } = props;
+
+  const ariaLabel = props['aria-label'];
```
Comment (khmakoto) [inline packages/react-next/src/components/Slider/useSlider.ts:29]: You can destructure this above by doing `'aria-label': ariaLabel.

Comment (joschect) [reply packages/react-next/src/components/Slider/useSlider.ts:29]: Neat! I didn't know that.

**Comment by joschect** [comment]
Any last minute comments? I'll complete the pull

**Comment by joschect** [comment]
Rebuilding since it's been a bit

**Pull request merged**